lib/esm-utils.js

```diff
@@ -30,14 +30,55 @@ const formattedImport = async file => {
   return import(file);
 };
 
-exports.requireOrImport = async file => {
+const hasStableEsmImplementation = (() => {
+  const [major, minor] = process.version.split('.');
+  // ESM is stable from v12.22.0 onward
+  // https://nodejs.org/api/esm.html#esm_modules_ecmascript_modules
+  return parseInt(major.slice(1), 10) > 12 || parseInt(minor, 10) >= 22;
+})();
+
+exports.requireOrImport = hasStableEsmImplementation
+  ? async file => {
+      if (path.extname(file) === '.mjs') {
+        return formattedImport(file);
+      }
+      try {
+        return dealWithExports(await formattedImport(file));
+      } catch (err) {
+        if (
+          err.code === 'ERR_MODULE_NOT_FOUND' ||
+          err.code === 'ERR_UNKNOWN_FILE_EXTENSION'
+        ) {
+          return require(file);
+        } else {
+          throw err;
+        }
+      }
+    }
+  : implemenetationOfRequireOrImportForUnstableEsm;
+
+function dealWithExports(module) {
+  if (module.default) {
+    return module.default;
+  } else {
+    return {...module, default: undefined};
+  }
+}
+
+exports.loadFilesAsync = async (files, preLoadFunc, postLoadFunc) => {
+  for (const file of files) {
+    preLoadFunc(file);
+    const result = await exports.requireOrImport(path.resolve(file));
+    postLoadFunc(file, result);
+  }
+};
+
+async function implemenetationOfRequireOrImportForUnstableEsm(file) {
```

Comment:
typo: implementationOf....

By inserting `/* istanbul ignore next */ this function should be ignored for coverage purposes. IMO we should do this, our CI tests will never take this path. We already use it in our code (eg. in _runner.js_).

Reply:
Done